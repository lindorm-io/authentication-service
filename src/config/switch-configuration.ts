import { IConfiguration } from "../typing";
import { NodeEnvironment } from "@lindorm-io/core";
import { configuration as development } from "./default-development";
import { configuration as production } from "./default-production";
import { configuration as staging } from "./default-staging";
import { configuration as test } from "./default-test";

export const switchConfiguration = (environment: string): IConfiguration => {
  switch (environment) {
    case NodeEnvironment.PRODUCTION:
      return production;

    case NodeEnvironment.STAGING:
      return staging;

    case NodeEnvironment.DEVELOPMENT:
      return development;

    case NodeEnvironment.TEST:
      return test;

    default:
      return development;
  }
};
