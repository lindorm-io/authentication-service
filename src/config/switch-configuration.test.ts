import { switchConfiguration } from "./switch-configuration";
import { NodeEnvironment } from "@lindorm-io/core";

describe("switchConfiguration", () => {
  test("should return PRODUCTION", () => {
    expect(switchConfiguration(NodeEnvironment.PRODUCTION)).toMatchSnapshot();
  });

  test("should return STAGING", () => {
    expect(switchConfiguration(NodeEnvironment.STAGING)).toMatchSnapshot();
  });

  test("should return DEVELOPMENT", () => {
    expect(switchConfiguration(NodeEnvironment.DEVELOPMENT)).toMatchSnapshot();
  });

  test("should return TEST", () => {
    expect(switchConfiguration(NodeEnvironment.TEST)).toMatchSnapshot();
  });
});
