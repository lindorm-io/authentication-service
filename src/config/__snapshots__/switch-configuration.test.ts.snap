// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`switchConfiguration should return DEVELOPMENT 1`] = `
Object {
  "ACCOUNT_OTP_ISSUER": "dev.lindorm.io",
  "CRYPTO_AES_SECRET": "secret",
  "CRYPTO_SHA_SECRET": "secret",
  "HOST": "http://localhost/",
  "JWT_ACCESS_TOKEN_EXPIRY": "1 minutes",
  "JWT_AUTHORIZATION_TOKEN_EXPIRY": "30 minutes",
  "JWT_IDENTITY_TOKEN_EXPIRY": "5 minutes",
  "JWT_ISSUER": "https://dev.lindorm.io/",
  "JWT_MULTI_FACTOR_TOKEN_EXPIRY": "30 minutes",
  "JWT_REFRESH_TOKEN_EXPIRY": "5 minutes",
  "MAILGUN_API_KEY": "api-key",
  "MAILGUN_DOMAIN": "dev.lindorm.io",
  "MAILGUN_FROM": "noreply@lindorm.io",
  "MONGO_DB_NAME": "authentication",
  "MONGO_EXPOSE_PORT": 27017,
  "MONGO_HOST": "localhost",
  "MONGO_INITDB_ROOT_PASSWORD": "password",
  "MONGO_INITDB_ROOT_USERNAME": "root",
  "NODE_ENVIRONMENT": "development",
  "SERVER_PORT": 3000,
}
`;

exports[`switchConfiguration should return PRODUCTION 1`] = `
Object {
  "ACCOUNT_OTP_ISSUER": "lindorm.io",
  "CRYPTO_AES_SECRET": null,
  "CRYPTO_SHA_SECRET": null,
  "HOST": "https://lindorm.io/",
  "JWT_ACCESS_TOKEN_EXPIRY": "3 minutes",
  "JWT_AUTHORIZATION_TOKEN_EXPIRY": "15 minutes",
  "JWT_IDENTITY_TOKEN_EXPIRY": "7 days",
  "JWT_ISSUER": "https://lindorm.io/",
  "JWT_MULTI_FACTOR_TOKEN_EXPIRY": "5 minutes",
  "JWT_REFRESH_TOKEN_EXPIRY": "7 days",
  "MAILGUN_API_KEY": null,
  "MAILGUN_DOMAIN": "lindorm.io",
  "MAILGUN_FROM": "noreply@lindorm.io",
  "MONGO_DB_NAME": null,
  "MONGO_EXPOSE_PORT": 0,
  "MONGO_HOST": null,
  "MONGO_INITDB_ROOT_PASSWORD": null,
  "MONGO_INITDB_ROOT_USERNAME": null,
  "NODE_ENVIRONMENT": "production",
  "SERVER_PORT": 3000,
}
`;

exports[`switchConfiguration should return STAGING 1`] = `
Object {
  "ACCOUNT_OTP_ISSUER": "staging.lindorm.io",
  "CRYPTO_AES_SECRET": null,
  "CRYPTO_SHA_SECRET": null,
  "HOST": "https://staging.lindorm.io/",
  "JWT_ACCESS_TOKEN_EXPIRY": "3 minutes",
  "JWT_AUTHORIZATION_TOKEN_EXPIRY": "15 minutes",
  "JWT_IDENTITY_TOKEN_EXPIRY": "7 days",
  "JWT_ISSUER": "https://staging.lindorm.io/",
  "JWT_MULTI_FACTOR_TOKEN_EXPIRY": "5 minutes",
  "JWT_REFRESH_TOKEN_EXPIRY": "7 days",
  "MAILGUN_API_KEY": null,
  "MAILGUN_DOMAIN": "staging.lindorm.io",
  "MAILGUN_FROM": "noreply@lindorm.io",
  "MONGO_DB_NAME": null,
  "MONGO_EXPOSE_PORT": 0,
  "MONGO_HOST": null,
  "MONGO_INITDB_ROOT_PASSWORD": null,
  "MONGO_INITDB_ROOT_USERNAME": null,
  "NODE_ENVIRONMENT": "staging",
  "SERVER_PORT": 3000,
}
`;

exports[`switchConfiguration should return TEST 1`] = `
Object {
  "ACCOUNT_OTP_ISSUER": "test.lindorm.io",
  "CRYPTO_AES_SECRET": "secret",
  "CRYPTO_SHA_SECRET": "secret",
  "HOST": "http://localhost/",
  "JWT_ACCESS_TOKEN_EXPIRY": "2 minutes",
  "JWT_AUTHORIZATION_TOKEN_EXPIRY": "45 minutes",
  "JWT_IDENTITY_TOKEN_EXPIRY": "1 days",
  "JWT_ISSUER": "https://test.lindorm.io/",
  "JWT_MULTI_FACTOR_TOKEN_EXPIRY": "30 minutes",
  "JWT_REFRESH_TOKEN_EXPIRY": "1 days",
  "MAILGUN_API_KEY": "api-key",
  "MAILGUN_DOMAIN": "test.lindorm.io",
  "MAILGUN_FROM": "noreply@lindorm.io",
  "MONGO_DB_NAME": "authentication",
  "MONGO_EXPOSE_PORT": 27017,
  "MONGO_HOST": "localhost",
  "MONGO_INITDB_ROOT_PASSWORD": "password",
  "MONGO_INITDB_ROOT_USERNAME": "root",
  "NODE_ENVIRONMENT": "development",
  "SERVER_PORT": 3000,
}
`;
